### ISSUE TITLE
Application error: a server-side exception has occurred (see the server logs for more information).

### ISSUE THREAD

**Issue by oliverlabs** [microsoft/azurechatgpt#48]
When I deploy an app to an Azure Web App using GitHub Actions, the Action succeeds and the deployment is there. 

However, once I authenticate, I get the following error:

`Application error: a server-side exception has occurred (see the server logs for more information).`
`Digest: 807578399`

The url is `https://chatomg-app-[REDACTED].azurewebsites.net/chat`

**Comment by Nettitude-MS** [comment]
I am seeing the same here: 

2023-08-04T13:26:23  Welcome, you are now connected to log-streaming service.
Starting Log Tail -n 10 of existing logs ----
/home/LogFiles/__lastCheckTime.txt  (https://[REDACTED].scm.azurewebsites.net/api/vfs/LogFiles/__lastCheckTime.txt)08/04/2023 12:14:12
/home/LogFiles/kudu/trace/36ecca4b6270-54a583d6-d700-4f5c-ba1e-548a8ed6c99d.txt  (https://[REDACTED].scm.azurewebsites.net/api/vfs/LogFiles/kudu/trace/36ecca4b6270-54a583d6-d700-4f5c-ba1e-548a8ed6c99d.txt)
2023-08-04T12:12:38  Startup Request, url: /api/settings, method: GET, type: request, pid: 86,1,14, ScmType: None
/home/LogFiles/kudu/trace/fd08e91a2b9c-d576cc91-951a-4522-97e9-aebb535ae520.txt  (https://[REDACTED].scm.azurewebsites.net/api/vfs/LogFiles/kudu/trace/fd08e91a2b9c-d576cc91-951a-4522-97e9-aebb535ae520.txt)
2023-08-04T13:26:23  Startup Request, url: /api/logstream/, method: GET, type: request, pid: 86,1,5, ScmType: None
/home/LogFiles/2023_08_04_ln0xsdlwk00000G_default_docker.log  (https://[REDACTED].scm.azurewebsites.net/api/vfs/LogFiles/2023_08_04_ln0xsdlwk00000G_default_docker.log)
2023-08-04T12:09:48.325789075Z     at Module._resolveFilename (node:internal/modules/cjs/loader:1075:15)
2023-08-04T12:09:48.325793575Z     at Module._load (node:internal/modules/cjs/loader:920:27)
2023-08-04T12:09:48.325797975Z     at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
2023-08-04T12:09:48.325815576Z     at node:internal/main/run_main_module:23:47 {
2023-08-04T12:09:48.325820376Z   code: 'MODULE_NOT_FOUND',
2023-08-04T12:09:48.325824776Z   requireStack: []
2023-08-04T12:09:48.325829076Z }
2023-08-04T12:09:48.326638200Z
2023-08-04T12:09:48.326673201Z Node.js v18.16.0

**Comment by oliverlabs** [comment]
I solved the issue. It was caused by the latest commits as someone tried to fix typos in the `AZURE_COSMOSEDB_URI` and `AZURE_COSMOSEDB_KEY` variable name. I reverted to the last commit on Aug 3 where this worked and it solved the issue.

**Comment by oliverlabs** [comment]
@TimArrow try updating your fork with the latest commits from the `main` branch and re-run the script. It seems you're still getting issues with connecting to your database.

**Comment by TimArrow** [comment]
I forked it today, so it should be up-to-date i checked the URI of my Comsos it is correct, also checked the key value which is also correct. I am new to this GitHub stuff trying to figure it out. Do you mean by re-run the script to run the workflow again?

**Comment by oliverlabs** [comment]
@TimArrow, yes, naturally, you need to redeploy the web app by manually triggering the GitHub Actions, also once done, make sure all the env variables defined [here](https://github.com/microsoft/azurechatgpt/blob/main/docs/7-environment-variables.md) are set on your Web App (or locally in a `env.local` file). 

Hope this helps!

**Comment by TimArrow** [comment]
That did not solve my issue.
Application error: a server-side exception has occurred (see the server logs for more information).
Digest: 3110068063

**Comment by TimArrow** [comment]
Update: Seems to be a permission issue. If i use my LiveID /Owner Subscription i can access the app, if i use a user originating from the tenant i receive the above error, if i use a guest user invited into my tenant it works ?!

**Comment by oliverlabs** [comment]
@TimArrow , check the settings of your Enterprise App Registration. You might need to provide a Global Admin permission to access user details. Also, check if you need to explicitly assign a user role to that app registration for users.
